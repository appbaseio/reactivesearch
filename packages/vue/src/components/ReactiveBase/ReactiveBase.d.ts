import * as types from '../../types.ts';

export interface ReactiveBaseProps {
	app?: string;
	children?: types.children;
	credentials?: string;
	as?: string;
	headers?: types.headers;
	queryParams?: string;
	theme?: types.style;
	themePreset?: types.themePreset;
	type?: string;
	url?: string;
	mapKey?: string;
	style?: types.style;
	className?: string;
	graphQLUrl?: string;
	transformResponse?: (...args: any[]) => any;
	transformRequest?: (...args: any[]) => any;
	getSearchParams?: () => string;
	setSearchParams?: (newURL: string) => void;
	initialQueriesSyncTime?: number;
	analyticsConfig?: types.analyticsConfig;
	reactivesearchAPIConfig?: types.appbaseConfig;
	initialState?: types.children;
	analytics?: boolean;
	endpoint?: types.endpointConfig;
	contextCollector?: any;
}

declare function ReactiveBase(props: ReactiveBaseProps): JSX.Element;

export default ReactiveBase;
