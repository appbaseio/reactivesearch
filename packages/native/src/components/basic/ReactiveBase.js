import React, { Component } from 'react';
import { createProvider } from 'react-redux';
import { Container } from 'native-base';
import Appbase from 'appbase-js';

import configureStore, { storeKey } from '@appbaseio/reactivecore';

import types from '@appbaseio/reactivecore/lib/utils/types';

import ThemeProvider from '../../theme/ThemeProvider';
import theme from '../../theme';

/* use a custom store key so reactivesearch does not interfere
   with a different redux store in a nested context */
const Provider = createProvider(storeKey);

// for network debugging while development
// GLOBAL.XMLHttpRequest = GLOBAL.originalXMLHttpRequest || GLOBAL.XMLHttpRequest;

class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.type = props.type ? props.type : '*';

		const credentials
			= props.url && props.url.trim() !== '' && !props.credentials ? null : props.credentials;

		const config = {
			url: props.url && props.url.trim() !== '' ? props.url : 'https://scalr.api.appbase.io',
			app: props.app,
			credentials,
			type: this.type,
		};

		const { headers = {} } = props;
		const appbaseRef = Appbase(config);
		this.store = configureStore({ config, appbaseRef, headers });
	}

	render() {
		return (
			<ThemeProvider theming={{ ...theme, ...this.props.theme }}>
				<Provider store={this.store}>
					<Container>{this.props.children}</Container>
				</Provider>
			</ThemeProvider>
		);
	}
}

ReactiveBase.propTypes = {
	type: types.string,
	url: types.string,
	credentials: types.string,
	headers: types.headers,
	app: types.stringRequired,
	theme: types.style,
	children: types.children,
};

export default ReactiveBase;
