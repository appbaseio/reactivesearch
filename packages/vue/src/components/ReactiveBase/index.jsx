import configureStore from '@appbaseio/reactivecore';
import VueTypes from 'vue-types';
import Appbase from 'appbase-js';
import Provider from '../Provider';
import { composeThemeObject } from '../../utils/index';
import types from '../../utils/vueTypes';
import URLParamsProvider from '../URLParamsProvider.jsx';
import getTheme from '../../styles/theme';

const URLSearchParams = require('url-search-params');

const ReactiveBase = {
	name: 'ReactiveBase',
	data() {
		this.state = {
			key: '__REACTIVE_BASE__',
		};
		return this.state;
	},
	created() {
		this.setStore(this.$props);
	},
	props: {
		app: types.stringRequired,
		analytics: VueTypes.bool.def(false),
		credentials: types.string,
		headers: types.headers,
		queryParams: types.string,
		theme: VueTypes.object.def({}),
		themePreset: VueTypes.string.def('light'),
		type: types.string,
		url: types.string,
		mapKey: types.string,
		className: types.string,
		initialState: VueTypes.object.def({}),
		transformRequest: types.func,
	},
	provide() {
		return {
			theme: composeThemeObject(getTheme(this.$props.themePreset), this.$props.theme),
			store: this.store,
		};
	},
	watch: {
		app() {
			this.updateState(this.$props);
		},
		url() {
			this.updateState(this.$props);
		},
		type() {
			this.updateState(this.$props);
		},
		credentials() {
			this.updateState(this.$props);
		},
		mapKey() {
			this.updateState(this.$props);
		},
		headers() {
			this.updateState(this.$props);
		},
	},
	methods: {
		updateState(props) {
			this.setStore(props);
			this.setState(state => ({
				key: `${state.key}-0`,
			}));
		},
		setStore(props) {
			const credentials
				= props.url && props.url.trim() !== '' && !props.credentials
					? null
					: props.credentials;
			const config = {
				url:
					props.url && props.url.trim() !== ''
						? props.url
						: 'https://scalr.api.appbase.io',
				app: props.app,
				credentials,
				type: props.type ? props.type : '*',
				transformRequest: props.transformRequest,
				analytics: props.analytics,
			};
			let queryParams = '';

			if (typeof window !== 'undefined') {
				queryParams = window.location.search;
			} else {
				queryParams = props.queryParams || '';
			}

			const params = new URLSearchParams(queryParams);
			let selectedValues = {};

			Array.from(params.keys()).forEach(key => {
				try {
					const parsedParams = JSON.parse(params.get(key));
					const selectedValue = {};
					if (parsedParams.value) {
						selectedValue.value = parsedParams.value;
					} else {
						selectedValue.value = parsedParams;
					}
					if (parsedParams.category) selectedValue.category = parsedParams.category;
					selectedValues = {
						...selectedValues,
						[key]: selectedValue,
					};
				} catch (e) {
					// Do not add to selectedValues if JSON parsing fails.
				}
			});

			const { headers = {}, themePreset } = props;
			const appbaseRef = Appbase(config);

			if (this.$props.transformRequest) {
				appbaseRef.transformRequest = this.$props.transformRequest;
			}

			const initialState = {
				config: {
					...config,
					mapKey: props.mapKey,
					themePreset,
				},
				appbaseRef,
				selectedValues,
				headers,
				...this.$props.initialState,
			};
			this.store = configureStore(initialState);
		},
	},
	render() {
		const children = this.$slots.default;
		const { headers, style, className } = this.$props;
		return (
			<Provider store={this.store}>
				<URLParamsProvider headers={headers} style={style} className={className}>
					{children}
				</URLParamsProvider>
			</Provider>
		);
	},
};
ReactiveBase.install = function(Vue) {
	Vue.component(ReactiveBase.name, ReactiveBase);
};

export default ReactiveBase;
