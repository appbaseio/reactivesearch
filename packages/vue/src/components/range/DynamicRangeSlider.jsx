import VueTypes from 'vue-types';
import NoSSR from 'vue-no-ssr';
import { Actions, helper } from '@appbaseio/reactivecore';
import { componentTypes } from '@appbaseio/reactivecore/lib/utils/constants';
import Container from '../../styles/Container';
import { connect, updateCustomQuery, getValidPropsKeys, isQueryIdentical } from '../../utils/index';
import Title from '../../styles/Title';
import Slider from '../../styles/Slider';
import types from '../../utils/vueTypes';
import { getComponents } from './addons/ssr';

const {
	addComponent,
	removeComponent,
	watchComponent,
	updateQuery,
	setQueryListener,
	setQueryOptions,
	setComponentProps,
	setCustomQuery,
	updateComponentProps,
} = Actions;

const {
	checkValueChange,
	getClassName,
	getOptionsFromQuery,
	isEqual,
	checkSomePropChange,
} = helper;

const DynamicRangeSlider = {
	name: 'DynamicRangeSlider',

	components: getComponents(),

	props: {
		beforeValueChange: types.func,
		className: VueTypes.string.def(''),
		rangeLabels: types.func,
		componentId: types.stringRequired,
		customQuery: types.func,
		data: types.data,
		dataField: types.stringRequired,
		defaultValue: types.func,
		filterLabel: types.string,
		innerClass: types.style,
		react: types.react,
		showFilter: VueTypes.bool.def(true),
		showCheckbox: VueTypes.bool.def(true),
		title: types.title,
		URLParams: VueTypes.bool.def(false),
		sliderOptions: VueTypes.object.def({}),
		nestedField: types.string,
	},

	data() {
		this.internalRangeComponent = `${this.$props.componentId}__range__internal`;

		return {
			currentValue: null,
			stats: [],
		};
	},

	created() {
		const onQueryChange = (...args) => {
			this.$emit('queryChange', ...args);
			this.$emit('query-change', ...args);
		};
		this.setQueryListener(this.$props.componentId, onQueryChange, null);
		// Update props in store
		this.setComponentProps(this.componentId, this.$props, componentTypes.dynamicRangeSlider);
		this.setComponentProps(
			this.internalRangeComponent,
			this.$props,
			componentTypes.dynamicRangeSlider,
		);
		// Set custom query in store
		updateCustomQuery(this.componentId, this.setCustomQuery, this.$props, this.currentValue);
	},
	mounted() {
		this.setReact();
	},
	beforeMount() {
		this.addComponent(this.$props.componentId);
		this.addComponent(this.internalRangeComponent);
		if (Array.isArray(this.selectedValue)) {
			this.handleChange(this.selectedValue);
		} else if (this.selectedValue) {
			this.handleChange(DynamicRangeSlider.parseValue(this.selectedValue, this.$props));
		}

		// get range before executing other queries
		this.updateRangeQueryOptions();
	},

	beforeUpdate() {
		if (!this.currentValue) {
			this.setDefaultValue(this.range);
		}
	},

	beforeDestroy() {
		this.removeComponent(this.$props.componentId);
		this.removeComponent(this.internalRangeComponent);
	},

	methods: {
		setDefaultValue({ start, end }) {
			if (this.$props.defaultValue) {
				const { start: defaultStart, end: defaultEnd } = this.$props.defaultValue(
					start,
					end,
				);
				this.handleChange([defaultStart, defaultEnd]);
			} else {
				this.currentValue = [start, end];
			}
		},

		setReact() {
			if (this.$props.react) {
				this.watchComponent(this.internalRangeComponent, this.$props.react);
				this.watchComponent(this.$props.componentId, this.$props.react);
			} else {
				this.watchComponent(this.internalRangeComponent, {});
				this.watchComponent(this.$props.componentId, {});
			}
		},

		rangeQuery() {
			return {
				min: { min: { field: this.$props.dataField } },
				max: { max: { field: this.$props.dataField } },
			};
		},

		updateRangeQueryOptions() {
			let aggs = {};

			if (this.$props.nestedField) {
				aggs = {
					[this.$props.nestedField]: {
						nested: {
							path: this.$props.nestedField,
						},
						aggs: this.rangeQuery(),
					},
				};
			} else {
				aggs = this.rangeQuery();
			}

			this.setQueryOptions(this.internalRangeComponent, { aggs });
		},

		handleSlider(values) {
			this.handleChange(values.currentValue);
		},

		handleChange(currentValue) {
			// Always keep the values within range
			const normalizedValue = [
				this.range ? Math.max(this.range.start, currentValue[0]) : currentValue[0],
				this.range ? Math.min(this.range.end, currentValue[1]) : currentValue[1],
			];

			const performUpdate = () => {
				this.currentValue = normalizedValue;
				this.updateQueryHandler(normalizedValue, this.$props);
				this.$emit('valueChange', { start: normalizedValue[0], end: normalizedValue[1] });
				this.$emit('value-change', { start: normalizedValue[0], end: normalizedValue[1] });
			};

			checkValueChange(
				this.$props.componentId,
				{
					start: normalizedValue[0],
					end: normalizedValue[1],
				},
				this.$props.beforeValueChange,
				performUpdate,
			);
		},

		updateQueryHandler(value) {
			let query = DynamicRangeSlider.defaultQuery(value, this.$props);
			let customQueryOptions;

			if (this.$props.customQuery) {
				({ query } = this.$props.customQuery(value, this.$props) || {});
				customQueryOptions = getOptionsFromQuery(
					this.$props.customQuery(value, this.$props),
				);
				updateCustomQuery(this.componentId, this.setCustomQuery, this.$props, value);
			}

			const { start, end } = this.range || { start: value[0], end: value[1] };
			const [currentStart, currentEnd] = value;
			// check if the slider is at its initial position
			const isInitialValue = currentStart === start && currentEnd === end;
			this.setQueryOptions(this.$props.componentId, customQueryOptions);

			this.updateQuery({
				componentId: this.$props.componentId,
				query,
				value,
				label: this.$props.filterLabel,
				showFilter: this.$props.showFilter && !isInitialValue,
				URLParams: this.$props.URLParams,
				componentType: componentTypes.dynamicRangeSlider,
			});
		},
	},

	computed: {
		labels() {
			if (!this.rangeLabels) return null;
			return this.rangeLabels(this.range.start, this.range.end);
		},
	},

	watch: {
		$props: {
			deep: true,
			handler(newVal) {
				const propsKeys = getValidPropsKeys(newVal);
				checkSomePropChange(newVal, this.componentProps, propsKeys, () => {
					this.updateComponentProps(
						this.componentId,
						newVal,
						componentTypes.dynamicRangeSlider,
					);
					this.updateComponentProps(
						this.internalRangeComponent,
						newVal,
						componentTypes.dynamicRangeSlider,
					);
				});
			},
		},
		react() {
			this.setReact();
		},

		selectedValue(newValue) {
			if (isEqual(newValue, this.currentValue)) return;

			const value = newValue || {
				start: this.range.start,
				end: this.range.end,
			};

			this.handleChange(DynamicRangeSlider.parseValue(value, this.$props));
		},

		range(newValue, oldValue) {
			if (isEqual(newValue, oldValue) || !this.currentValue) return;

			const [currentStart, currentEnd] = this.currentValue || [];
			const { start: oldStart, end: oldEnd } = oldValue || {};

			const newStart = currentStart === oldStart ? newValue.start : currentStart;
			const newEnd = currentEnd === oldEnd ? newValue.end : currentEnd;

			this.handleChange([newStart, newEnd]);
		},
		customQuery(newVal, oldVal) {
			if (!isQueryIdentical(newVal, oldVal, this.$data.currentValue, this.$props)) {
				this.updateQueryHandler(this.$data.currentValue);
			}
		},
	},

	render() {
		if (!this.range) {
			return null;
		}
		const { start, end } = this.range;
		return (
			<Container class={this.$props.className}>
				{this.$props.title && (
					<Title class={getClassName(this.$props.innerClass, 'title')}>
						{this.$props.title}
					</Title>
				)}
				<NoSSR>
					<Slider class={getClassName(this.$props.innerClass, 'slider')}>
						<vue-slider-component
							value={[
								Math.max(start, this.currentValue[0]),
								Math.min(end, this.currentValue[1]),
							]}
							min={Math.min(start, this.currentValue[0])}
							max={Math.max(end, this.currentValue[1])}
							onDrag-end={this.handleSlider}
							dotSize={20}
							height={4}
							enable-cross={false}
							{...{ props: this.$props.sliderOptions }}
						/>

						{this.labels ? (
							<div class="label-container">
								<label
									class={
										getClassName(this.$props.innerClass, 'label')
										|| 'range-label-left'
									}
								>
									{this.labels.start}
								</label>
								<label
									class={
										getClassName(this.$props.innerClass, 'label')
										|| 'range-label-right'
									}
								>
									{this.labels.end}
								</label>
							</div>
						) : null}
					</Slider>
				</NoSSR>
			</Container>
		);
	},
};

DynamicRangeSlider.defaultQuery = (values, props) => {
	let query = null;

	if (Array.isArray(values) && values.length) {
		query = {
			range: {
				[props.dataField]: {
					gte: values[0],
					lte: values[1],
					boost: 2.0,
				},
			},
		};
	}

	if (query && props.nestedField) {
		return {
			query: {
				nested: {
					path: props.nestedField,
					query,
				},
			},
		};
	}

	return query;
};

DynamicRangeSlider.parseValue = value => [value.start, value.end];

const mapStateToProps = (state, props) => {
	const componentId = state.aggregations[props.componentId];
	const internalRange = state.aggregations[`${props.componentId}__range__internal`];

	let options = componentId && componentId[props.dataField];
	let range = state.aggregations[`${props.componentId}__range__internal`];

	if (props.nestedField) {
		options
			= options
			&& componentId[props.dataField][props.nestedField]
			&& componentId[props.dataField][props.nestedField].buckets
				? componentId[props.dataField][props.nestedField].buckets
				: [];
		range
			= range && internalRange[props.nestedField].min
				? {
					start: internalRange[props.nestedField].min.value,
					end: internalRange[props.nestedField].max.value,
				  }
				: null;
	} else {
		options
			= options && componentId[props.dataField].buckets
				? componentId[props.dataField].buckets
				: [];
		range
			= range && internalRange.min
				? { start: internalRange.min.value, end: internalRange.max.value }
				: null;
	}

	return {
		options,
		range,
		selectedValue: state.selectedValues[props.componentId]
			? state.selectedValues[props.componentId].value
			: null,
		componentProps: state.props[props.componentId],
	};
};

const mapDispatchtoProps = {
	addComponent,
	removeComponent,
	updateQuery,
	watchComponent,
	setQueryListener,
	setQueryOptions,
	setComponentProps,
	setCustomQuery,
	updateComponentProps,
};

const RangeConnected = connect(mapStateToProps, mapDispatchtoProps)(DynamicRangeSlider);

DynamicRangeSlider.install = function(Vue) {
	Vue.component(DynamicRangeSlider.name, RangeConnected);
};

// Add componentType for SSR
DynamicRangeSlider.componentType = componentTypes.dynamicRangeSlider;

export default DynamicRangeSlider;
