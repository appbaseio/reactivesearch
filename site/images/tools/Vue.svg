<?xml version="1.0" encoding="UTF-8"?>
<svg width="142px" height="117px" viewBox="0 0 142 117" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: sketchtool 52.4 (67378) - http://www.bohemiancoding.com/sketch -->
    <title>F5968638-E0A9-437B-8498-2A31D8EDA594@3x</title>
    <desc>Created with sketchtool.</desc>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Reactive-Search-//-Tools" transform="translate(-801.000000, -485.000000)" fill-rule="nonzero" stroke="#FFFFFF">
            <g id="Group-5" transform="translate(784.000000, 488.000000)">
                <g id="Group-13" transform="translate(20.000000, 0.000000)">
                    <polygon id="Path-4" stroke-width="3" opacity="0.46765718" points="4.40536496e-13 -1.42108547e-14 52.5512501 -1.42108547e-14 67.7970088 25.5378091 83.1987012 -1.42108547e-14 135.434835 -1.42108547e-14 67.7174177 111"></polygon>
                    <path d="M28,-1.49213975e-13 L52.2655434,-1.49213975e-13 L67.2351697,25.5378091 L83.4652594,-1.49213975e-13 C99.2751946,-1.49213975e-13 107.180162,-1.49213975e-13 107.180162,-1.49213975e-13 C107.180162,-1.49213975e-13 93.9834685,22.8776207 67.5900811,68.632862 L28,-1.49213975e-13 Z" id="Path-15" stroke-width="5"></path>
                </g>
            </g>
        </g>
    </g>
</svg>