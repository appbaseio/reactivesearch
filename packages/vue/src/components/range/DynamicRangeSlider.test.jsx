import { mount } from '@vue/test-utils';
import Vue from 'vue';
import { RangeConnected as DynamicRangeSlider } from './DynamicRangeSlider.jsx';
import ReactiveBase from '../ReactiveBase/index.jsx';

Vue.component('vue-slider-component', require('vue-slider-component'));

const MOCK_RANGE_DATA = {
	min: {
		value: 1,
	},
	max: {
		value: 3455,
	},
};

describe('DynamicRangeSlider', () => {
	it('should render range with slider', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<DynamicRangeSlider
							mode="test"
							componentId="DynamicRangeMock"
							dataField="books_count"
							mockData={{
								DynamicRangeMock__range__internal: {
									aggregations: MOCK_RANGE_DATA,
								},
							}}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should render title', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<DynamicRangeSlider
							mode="test"
							componentId="DynamicRangeMock"
							dataField="books_count"
							mockData={{
								DynamicRangeMock__range__internal: {
									aggregations: MOCK_RANGE_DATA,
								},
							}}
							title="Test Title"
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should render range labels', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<DynamicRangeSlider
							mode="test"
							componentId="DynamicRangeMock"
							dataField="books_count"
							mockData={{
								DynamicRangeMock__range__internal: {
									aggregations: MOCK_RANGE_DATA,
								},
							}}
							rangeLabels={(start, end) => ({
								start,
								end,
							})}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should select default value', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<DynamicRangeSlider
							mode="test"
							componentId="DynamicRangeMock"
							dataField="books_count"
							mockData={{
								DynamicRangeMock__range__internal: {
									aggregations: MOCK_RANGE_DATA,
								},
							}}
							defaultValue={() => ({
								start: 30,
								end: 2300,
							})}
							rangeLabels={() => ({
								start: 30,
								end: 2300,
							})}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});

	it('should not display tooltip when sliderOptions has tooltip set to false', () => {
		const wrapper = mount({
			name: 'TestComponent',
			render() {
				return (
					<ReactiveBase app="test" url="https://foo:bar@localhost:800">
						<DynamicRangeSlider
							mode="test"
							componentId="DynamicRangeMock"
							dataField="books_count"
							mockData={{
								DynamicRangeMock__range__internal: {
									aggregations: MOCK_RANGE_DATA,
								},
							}}
							defaultValue={() => ({
								start: 30,
								end: 2300,
							})}
							rangeLabels={() => ({
								start: 30,
								end: 2300,
							})}
							sliderOptions={{
								tooltip: false,
							}}
						/>
					</ReactiveBase>
				);
			},
		});

		expect(wrapper.element).toMatchSnapshot();
	});
});
