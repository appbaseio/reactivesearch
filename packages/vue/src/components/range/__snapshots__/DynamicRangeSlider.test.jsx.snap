// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicRangeSlider should not display tooltip when sliderOptions has tooltip set to false 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
  >
    <div
      class="css-1sjm8x1"
    >
      <div
        class="vue-slider vue-slider-ltr"
        style="padding: 10px 0px; width: auto; height: 4px;"
        tooltip="always"
      >
        <div
          class="vue-slider-rail"
        >
          <div
            class="vue-slider-process"
            style="height: 100%; top: 0px; left: 0.8396062536189924%; width: 65.72090330052113%; transition-property: width,left; transition-duration: 0.5s;"
          />
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="30"
            aria-valuetext="30"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 0.8396062536189924%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  30
                </span>
              </div>
            </div>
          </div>
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="2300"
            aria-valuetext="2300"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 66.56050955414013%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  2300
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="label-container"
      >
        <label
          class="range-label-left"
        >
          30
        </label>
        <label
          class="range-label-right"
        >
          2300
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`DynamicRangeSlider should render range labels 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
  >
    <div
      class="css-1sjm8x1"
    >
      <div
        class="vue-slider vue-slider-ltr"
        style="padding: 10px 0px; width: auto; height: 4px;"
      >
        <div
          class="vue-slider-rail"
        >
          <div
            class="vue-slider-process"
            style="height: 100%; top: 0px; left: 0%; width: 100%; transition-property: width,left; transition-duration: 0.5s;"
          />
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="1"
            aria-valuetext="1"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 0%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  1
                </span>
              </div>
            </div>
          </div>
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="3455"
            aria-valuetext="3455"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 100%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  3455
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="label-container"
      >
        <label
          class="range-label-left"
        >
          1
        </label>
        <label
          class="range-label-right"
        >
          3455
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`DynamicRangeSlider should render range with slider 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
  >
    <div
      class="css-1sjm8x1"
    >
      <div
        class="vue-slider vue-slider-ltr"
        style="padding: 10px 0px; width: auto; height: 4px;"
      >
        <div
          class="vue-slider-rail"
        >
          <div
            class="vue-slider-process"
            style="height: 100%; top: 0px; left: 0%; width: 100%; transition-property: width,left; transition-duration: 0.5s;"
          />
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="1"
            aria-valuetext="1"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 0%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  1
                </span>
              </div>
            </div>
          </div>
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="3455"
            aria-valuetext="3455"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 100%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  3455
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DynamicRangeSlider should render title 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
  >
    <h2
      class="css-dysiaa"
    >
      Test Title
    </h2>
    <div
      class="css-1sjm8x1"
    >
      <div
        class="vue-slider vue-slider-ltr"
        style="padding: 10px 0px; width: auto; height: 4px;"
      >
        <div
          class="vue-slider-rail"
        >
          <div
            class="vue-slider-process"
            style="height: 100%; top: 0px; left: 0%; width: 100%; transition-property: width,left; transition-duration: 0.5s;"
          />
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="1"
            aria-valuetext="1"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 0%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  1
                </span>
              </div>
            </div>
          </div>
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="3455"
            aria-valuetext="3455"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 100%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  3455
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DynamicRangeSlider should select default value 1`] = `
<div
  class="css-nw180v"
>
  <div
    class="css-m1gst5"
  >
    <div
      class="css-1sjm8x1"
    >
      <div
        class="vue-slider vue-slider-ltr"
        style="padding: 10px 0px; width: auto; height: 4px;"
      >
        <div
          class="vue-slider-rail"
        >
          <div
            class="vue-slider-process"
            style="height: 100%; top: 0px; left: 0.8396062536189924%; width: 65.72090330052113%; transition-property: width,left; transition-duration: 0.5s;"
          />
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="30"
            aria-valuetext="30"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 0.8396062536189924%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  30
                </span>
              </div>
            </div>
          </div>
          <div
            aria-orientation="horizontal"
            aria-valuemax="3455"
            aria-valuemin="1"
            aria-valuenow="2300"
            aria-valuetext="2300"
            class="vue-slider-dot"
            role="slider"
            style="width: 20px; height: 20px; transform: translate(-50%, -50%); top: 50%; left: 66.56050955414013%; transition: left 0.5s;"
            tabindex="0"
          >
            <div
              class="vue-slider-dot-handle"
            />
            <div
              class="vue-slider-dot-tooltip vue-slider-dot-tooltip-top vue-slider-dot-tooltip-show"
            >
              <div
                class="vue-slider-dot-tooltip-inner vue-slider-dot-tooltip-inner-top"
              >
                <span
                  class="vue-slider-dot-tooltip-text"
                >
                  2300
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="label-container"
      >
        <label
          class="range-label-left"
        >
          30
        </label>
        <label
          class="range-label-right"
        >
          2300
        </label>
      </div>
    </div>
  </div>
</div>
`;
